drivers/timer/xlnx_psttc_timer.c

```diff
@@ -1,187 +1,186 @@
-	z_clock_announce(_sys_idle_elapsed_ticks);
+	/* Advance tick count by 1 */
+	ticks = 1;
+#endif
+
+	/* Announce to the kernel*/
+	z_clock_announce(ticks);
 }
 
 int z_clock_driver_init(struct device *device)
 {
-	int ret;
-	u32_t interval;
-	u8_t prescaler;
-	u32_t regval;
+	u32_t reg_val;
 
 	/* Stop timer */
 	sys_write32(XTTCPS_CNT_CNTRL_DIS_MASK,
-		    TIMER_BASEADDR + XTTCPS_CNT_CNTRL_OFFSET);
+		    TIMER_BASE_ADDR + XTTCPS_CNT_CNTRL_OFFSET);
 
-	/* Calculate prescaler */
-	ret = xttc_calculate_interval(&interval, &prescaler);
-	if (ret < 0) {
-		printk("Failed to calculate prescaler.\n");
-		return ret;
-	}
+#ifdef CONFIG_TICKLESS_KERNEL
+	/* Initialise internal states */
+	last_cycles = 0;
+#endif
 
-	/* Reset registers */
+	/* Initialise timer registers */
 	sys_write32(XTTCPS_CNT_CNTRL_RESET_VALUE,
-		    TIMER_BASEADDR + XTTCPS_CNT_CNTRL_OFFSET);
-	sys_write32(0, TIMER_BASEADDR + XTTCPS_CLK_CNTRL_OFFSET);
-	sys_write32(0, TIMER_BASEADDR + XTTCPS_INTERVAL_VAL_OFFSET);
-	sys_write32(0, TIMER_BASEADDR + XTTCPS_MATCH_0_OFFSET);
-	sys_write32(0, TIMER_BASEADDR + XTTCPS_MATCH_1_OFFSET);
-	sys_write32(0, TIMER_BASEADDR + XTTCPS_MATCH_2_OFFSET);
-	sys_write32(0, TIMER_BASEADDR + XTTCPS_IER_OFFSET);
-	sys_write32(XTTCPS_IXR_ALL_MASK, TIMER_BASEADDR + XTTCPS_ISR_OFFSET);
+		    TIMER_BASE_ADDR + XTTCPS_CNT_CNTRL_OFFSET);
+	sys_write32(0, TIMER_BASE_ADDR + XTTCPS_CLK_CNTRL_OFFSET);
+	sys_write32(0, TIMER_BASE_ADDR + XTTCPS_INTERVAL_VAL_OFFSET);
+	sys_write32(0, TIMER_BASE_ADDR + XTTCPS_MATCH_0_OFFSET);
+	sys_write32(0, TIMER_BASE_ADDR + XTTCPS_MATCH_1_OFFSET);
+	sys_write32(0, TIMER_BASE_ADDR + XTTCPS_MATCH_2_OFFSET);
+	sys_write32(0, TIMER_BASE_ADDR + XTTCPS_IER_OFFSET);
+	sys_write32(XTTCPS_IXR_ALL_MASK, TIMER_BASE_ADDR + XTTCPS_ISR_OFFSET);
 
 	/* Reset counter value */
-	regval = sys_read32(TIMER_BASEADDR + XTTCPS_CNT_CNTRL_OFFSET);
-	regval |= XTTCPS_CNT_CNTRL_RST_MASK;
-	sys_write32(regval, TIMER_BASEADDR + XTTCPS_CNT_CNTRL_OFFSET);
-
-	/* Set options */
-	regval = sys_read32(TIMER_BASEADDR + XTTCPS_CNT_CNTRL_OFFSET);
-	regval |= XTTCPS_CNT_CNTRL_INT_MASK;
-	sys_write32(regval, TIMER_BASEADDR + XTTCPS_CNT_CNTRL_OFFSET);
-
-	/* Set interval and prescaller */
-	sys_write32(interval, TIMER_BASEADDR + XTTCPS_INTERVAL_VAL_OFFSET);
-	regval = (u32_t)((prescaler & 0xFU) << 1);
-	sys_write32(regval, TIMER_BASEADDR + XTTCPS_CLK_CNTRL_OFFSET);
+	reg_val = sys_read32(TIMER_BASE_ADDR + XTTCPS_CNT_CNTRL_OFFSET);
+	reg_val |= XTTCPS_CNT_CNTRL_RST_MASK;
+	sys_write32(reg_val, TIMER_BASE_ADDR + XTTCPS_CNT_CNTRL_OFFSET);
+
+	/* Set match mode */
+	reg_val = sys_read32(TIMER_BASE_ADDR + XTTCPS_CNT_CNTRL_OFFSET);
+	reg_val |= XTTCPS_CNT_CNTRL_MATCH_MASK;
+	sys_write32(reg_val, TIMER_BASE_ADDR + XTTCPS_CNT_CNTRL_OFFSET);
+
+	/* Set initial timeout */
+	reg_val = IS_ENABLED(CONFIG_TICKLESS_KERNEL) ?
+			CYCLES_NEXT_MAX : CYCLES_PER_TICK;
+	sys_write32(reg_val, TIMER_BASE_ADDR + XTTCPS_MATCH_0_OFFSET);
+
+	/* Connect timer interrupt */
```

Comment:
Style: "IRQ_CONNECT(TIMER_IRQ, ..." sufficiently implies that the timer interrupt is connected without the comment. :)

Reply:
Just wanted to provide a logical flow of this function in the comments (i.e. even without reading actual code, you will know what this function does just only reading comments).

Though, I agree that this would often qualify as a "state-the-obvious" kind of comment :p